### PR TITLE
19 scene editor

### PR DESCRIPTION
**Pull request by Skiper23**
modeling map grid, rotation of the selected image, modify the size of the map grid

### CODE REVIEW

File: Car Simulator/Assets/Scripts/MapValueDisplay.cs [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    private const string DEFAULT_IMAGE_NAME = "0_empty.png";
+    private int GRID_WIDTH = 5;
+    private  int GRID_HEIGHT = 5;
+    private const float OFFSET = 1.0f;
+    private const float CELL_WIDTH = 50.0f;
+    private const float CELL_HEIGHT = 50.0f;
+    private const float DEFAULT_ROTATE_ANGLE = 0.0f;
+    private const bool HEIGHT = true;
+    private const bool WIDTH = false;
+    private List<List<Tuple<string,float>>> mapGrid = new List<List<Tuple<string,float>>>();
```
Comment (lukrzak) [inline Car Simulator/Assets/Scripts/MapValueDisplay.cs:22]: Change to 2d string array

File: Car Simulator/Assets/Scripts/MapValueDisplay.cs [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+using System.Collections.Generic;
+public class MapValueDisplay : MonoBehaviour
+{
+    public GameObject imagePrefab;
+    public Image selectedImage;
+
+    private const string IMAGES_DIRECTORY_PATH = "Assets/Resources/PaletteImages/";
+    private const string DEFAULT_IMAGE_NAME = "0_empty.png";
+    private int GRID_WIDTH = 5;
+    private  int GRID_HEIGHT = 5;
```
Comment (lukrzak) [inline Car Simulator/Assets/Scripts/MapValueDisplay.cs:15]: Remove additional space

File: Car Simulator/Assets/Scripts/MapValueDisplay.cs [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+    private List<List<Tuple<string,float>>> mapGrid = new List<List<Tuple<string,float>>>();
+    void Start()
+    {
+        InicjalizeDefaultMapGrid();
+        SizeInputScript.onEndEdit += ModifySizeOfGrid;
+        GenerateGrid();
+    }
+    public void InicjalizeDefaultMapGrid()
+    {
+        for(int i=0;i<GRID_WIDTH; i++)
```
Comment (lukrzak) [inline Car Simulator/Assets/Scripts/MapValueDisplay.cs:31]: Add spaces between semicolons

File: Car Simulator/Assets/Scripts/MapValueDisplay.cs [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+            for (int j=0;j<GRID_HEIGHT; j++)
+            {
+                tmp.Add(new Tuple<string,float>(DEFAULT_IMAGE_NAME, DEFAULT_ROTATE_ANGLE));
+            }
+            mapGrid.Add(tmp);
+        }
+    }
+    public void ModifySizeOfGrid(string input, bool dimensionType)
+    {
+        int size=Int32.Parse(input);
```
Comment (lukrzak) [inline Car Simulator/Assets/Scripts/MapValueDisplay.cs:43]: Also add space

File: Car Simulator/Assets/Scripts/MapValueDisplay.cs [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+        DeleyGrid();
```
Comment (lukrzak) [inline Car Simulator/Assets/Scripts/MapValueDisplay.cs:44]: probably DeleteGrid()

File: Car Simulator/Assets/Scripts/MapValueDisplay.cs [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+            }
+            else
+            {
+                RemoveColumns(GRID_WIDTH - size);
+            }
+            GRID_WIDTH = size;
+        }
+        GenerateGrid();
+
+    }
```
Comment (lukrzak) [inline Car Simulator/Assets/Scripts/MapValueDisplay.cs:71]: Add separate line between functions

File: Car Simulator/Assets/Scripts/MapValueDisplay.cs [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+                AddColumns(size - GRID_WIDTH);
+            }
+            else
+            {
+                RemoveColumns(GRID_WIDTH - size);
+            }
+            GRID_WIDTH = size;
+        }
+        GenerateGrid();
+
```
Comment (lukrzak) [inline Car Simulator/Assets/Scripts/MapValueDisplay.cs:70]: Unnecessary empty line

File: Car Simulator/Assets/Scripts/MapValueDisplay.cs [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+        string imageName;
+
+        for (int y = 0; y < GRID_HEIGHT; y++)
+        {
+            for (int x = 0; x < GRID_WIDTH; x++)
+            {
+                float posX = (x * CELL_WIDTH) + (x * OFFSET);
+                float posY = (-y * CELL_HEIGHT) - (y * OFFSET);
+                imageName = IMAGES_DIRECTORY_PATH + mapGrid[y][x].Item1;
```
Comment (lukrzak) [inline Car Simulator/Assets/Scripts/MapValueDisplay.cs:127]: is Item1 built-in property?

File: Car Simulator/Assets/Scripts/MapValueDisplay.cs [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+
+                DisplayPaletteValueImage(posX, posY, imageName, imageIndex);
+                imageIndex++;
+            }
+        }
+    }
+    void DisplayPaletteValueImage(float posX, float posY, string imagePath,int imageIndex)
```
Comment (lukrzak) [inline Car Simulator/Assets/Scripts/MapValueDisplay.cs:134]: add visibility modifiers (preferably private)

File: Car Simulator/Assets/Scripts/SizeInputScript.cs [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+
+    void OnEndEditListener(string input)
+    {
+        bool dimensionType;
+        if (!int.TryParse(input, out int numericValue) || numericValue < 1)
+        {
+            inputField.text = "1";
+            numericValue = 1;
+        }
+        if(inputField.name=="Width Input")
```
Comment (lukrzak) [inline Car Simulator/Assets/Scripts/SizeInputScript.cs:30]: Add spaces

File: Car Simulator/Assets/Scripts/MapValueDisplay.cs [hunk 11]
```diff
@@ -0,0 +1,197 @@
+using System.IO;
+using System;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.EventSystems;
+using System.Collections.Generic;
```
Comment (lukrzak) [inline Car Simulator/Assets/Scripts/MapValueDisplay.cs:6]: Add empty line

File: Car Simulator/Assets/Scripts/MapValueDisplay.cs [hunk 12]
```diff
@@ -0,0 +1,197 @@
...
+public class MapValueDisplay : MonoBehaviour
+{
+    public GameObject imagePrefab;
+    public Image selectedImage;
+
+    private const string IMAGES_DIRECTORY_PATH = "Assets/Resources/PaletteImages/";
+    private const string DEFAULT_IMAGE_NAME = "0_empty.png";
+    private int GRID_WIDTH = 5;
```
Comment (lukrzak) [inline Car Simulator/Assets/Scripts/MapValueDisplay.cs:14]: Add const modifier

File: Car Simulator/Assets/Scripts/SizeInputScript.cs [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+
+public class SizeInputScript : MonoBehaviour
+{
+    public TMP_InputField inputField;
+    public TMPro.TextMeshProUGUI errorText;
+    private const bool HEIGHT = true;
+    private const bool WIDTH = false;
+    private const int MAX_HEIGHT = 10;
+    private const int MAX_WIDTH = 20;
+    public static event Action<string,bool> onEndEdit;
```
Comment (lukrzak) [inline Car Simulator/Assets/Scripts/SizeInputScript.cs:14]: Add space

File: Car Simulator/Assets/Scripts/SizeInputScript.cs [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+
+    void Start()
+    {
+        inputField.text = "5";
```
Comment (lukrzak) [inline Car Simulator/Assets/Scripts/SizeInputScript.cs:18]: Change magic number to soething like DEFAULT_SIZE

File: Car Simulator/Assets/Scripts/RotateSelectedImageScript.cs [hunk 15]
```diff
@@ -0,0 +1,24 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class RotateScript : MonoBehaviour
+{
+    public Image selectedImage;
+    int i = 0;
```
Comment (lukrzak) [inline Car Simulator/Assets/Scripts/RotateSelectedImageScript.cs:9]: is it used?

**Comment by Skiper23** [comment]
changed map Grid from 2d tuple list to 2d tuple array and fixed stylistic

**Review by lukrzak — APPROVED** [review]
lgtm

**Pull request merged**